app/components/search/GeoSearch.tsx

```diff
@@ -33,30 +40,88 @@ const GeoSearch = () => {
       }
     }
 
-    if (status === "idle") {
-      if (map?.getLayer("hits")) map.removeLayer("hits");
-      if (map?.getSource("hits")) map.removeSource("hits");
-
-      map?.addSource("hits", {
-        type: "geojson",
-        data: hitsToFeatureCollection(renderState.gca.hits.items),
-      });
-
-      map?.addLayer({
-        id: "hits",
-        type: "symbol",
-        source: "hits",
-        layout: {
-          "icon-image": "pulsing-dot",
-        },
-        filter: ["==", "$type", "Point"],
-      });
+    if (map.getLayer("hits")) map.removeLayer("hits");
+    if (map.getSource("hits")) map.removeSource("hits");
+
+    map.addSource("hits", {
+      type: "geojson",
+      data: hitsToFeatureCollection(hits),
+    });
+
+    map.addLayer({
+      id: "hits",
+      type: "symbol",
+      source: "hits",
+      layout: {
+        "icon-image": "pulsing-dot",
+      },
+      filter: ["==", "$type", "Point"],
+    });
+
+    const handleUnclusteredPointClick = (e: MapLayerMouseEvent) => {
+      if (!e.features || !e.features.length) return;
+
+      const feature = e.features[0];
+      console.log("Clicked Feature:", feature);
+
+      const clickedHit = hits.find(
+        (hit) => hit.identifier === feature.properties.identifier
+      );
+
+      console.log("Clicked Hit:", clickedHit);
+
+      if (clickedHit) {
+        const clickedPlace: TPlaceRecord = {
+          uuid: clickedHit.uuid,
+          name: clickedHit.name,
+          description: clickedHit.description,
+          place_names: clickedHit.place_names || [],
+          place_layers: clickedHit.place_layers || [],
+          web_identifiers: clickedHit.web_identifiers || [],
+          place_geometry: clickedHit.place_geometry || null,
```

Comment:
Your main problem is here. `clickedHit` does not have `place_geometry`, it has `geometry`. This line should be

~~~typescript
place_geometry: { geometry_json: clickedHit.geometry } || null,
~~~

That will give you a pop up, however, it does not seem to open the correct place. See note for line 68 above.